ci(docs-infra): run tests against local Angular packages too

gkalpak opened the pull request:
This allows catching regressions earlier.

gkalpak committed:
ci(docs-infra): remove unnecessary command

This was accidentally merged with 4d506acba and 87f60bccf.
The build script is called in `scripts/ci/build.sh` (if necessary).

gkalpak committed:
build(docs-infra): upgrade webpack-cli to 3.1.2

This is necessary to avoid webpack/webpack#8082, when installing
dependencies without taking the lockfile into account (e.g. with
`yarn aio-use-local` - locally or on CI).

mary-poppins commented:
You can preview 1dd3f31 at https://pr26202-1dd3f31.ngbuilds.io/.

mary-poppins commented:
You can preview 4353aa7 at https://pr26202-4353aa7.ngbuilds.io/.

mary-poppins commented:
You can preview a6e89a1 at https://pr26202-a6e89a1.ngbuilds.io/.

gkalpak committed:
test(docs-infra): fix tests

mary-poppins commented:
You can preview db28ec1 at https://pr26202-db28ec1.ngbuilds.io/.

mary-poppins commented:
You can preview 0a79555 at https://pr26202-0a79555.ngbuilds.io/.

mary-poppins commented:
You can preview e3a9c05 at https://pr26202-e3a9c05.ngbuilds.io/.

petebacondarwin commented on aio/src/app/custom-elements/contributor/contributor.service.spec.ts:
```diff
@@ -44,7 +44,7 @@ describe('ContributorService', () => {
       let completed = false;
       contribService.contributors.subscribe(undefined, undefined, () => completed = true);
-      expect(true).toBe(true, 'observable completed');
+      expect(completed).toBe(true, 'observable completed');
```
❤️

petebacondarwin commented on scripts/ci/install.sh:
```diff
@@ -49,15 +49,13 @@ travisFoldEnd "bower-install"
     (
-      # HACK (don't submit with this): Build Angular
-      ./build.sh --packages=compiler,core,elements --examples=false
-
```
LOL

IgorMinar replied:
omg! :)

petebacondarwin commented on aio/src/app/shared/copier.service.ts:
```diff
@@ -34,7 +35,7 @@ export class CopierService {
 
       // Move element to the same position vertically
-      const yPosition = window.pageYOffset || document.documentElement.scrollTop;
+      const yPosition = window.pageYOffset || docElem.scrollTop;
```
I get the feeling that this change should be in a different commit ... 🤔

gkalpak replied:
👍 Agreed :grin:
For the record, this was failing because of the updated `lib.dom.d.ts` typings in latest TS (which is brought in with `aio-use-local`).

gkalpak replied:
I moved it to a new commit, then it felt appropriate to upgrade TS to 3.1.x (which this change is related to), which in turn required upgrading Angular to 7.0.0-rc.0 :grin:

Now you have to review again 😛

petebacondarwin reviewed (approved):
LGTM - but I wonder if we should be adding new jobs to CircleCI rather than Travis?
Or are we just not in a position to build AIO via CircleCI yet?

gkalpak commented:
WRT Travis vs CircleCI:
This new job is very similar to the `aio` job. I think it makes sense to keep them on the same CI.
(Off the top of my head, there shouldn't be anything blocking us from moving the angular.io-related jobs to CircleCI.)

gkalpak committed:
ci(docs-infra): run tests against local Angular packages too

gkalpak committed:
build(docs-infra): upgrade Angular to 7.0.0-rc.0 and TypeScript to 3.1.1

gkalpak commented:
Rebased on master and added a new commit (per https://github.com/angular/angular/pull/26202/commits/b322d4c26baa821bf7e1dc1e5cbafcb8e7df446f#r222719118 - also took the chance to upgrade Angular and TS).

@petebacondarwin, @IgorMinar: Please, re-approve 🙏 (FWIW, only the last commit is new.)

mary-poppins commented:
You can preview 48fade2 at https://pr26202-48fade2.ngbuilds.io/.

mary-poppins commented:
You can preview c18a77c at https://pr26202-c18a77c.ngbuilds.io/.

jasonaden commented:
@gkalpak Can you create a separate PR for patch? Specifically because of the changes to `package.json` and `yarn.lock` as these will never merge cleanly.

mary-poppins commented:
You can preview 7b34786 at https://pr26202-7b34786.ngbuilds.io/.

gkalpak commented:
Rebased this on master and changed PR target to `master-only`.
Created #26306 for patch.
